sdk/src/main/java/io/dapr/config/Properties.java

```diff
@@ -172,4 +173,39 @@ public class Properties {
           "dapr.http.client.maxIdleConnections",
           "DAPR_HTTP_CLIENT_MAX_IDLE_CONNECTIONS",
           DEFAULT_HTTP_CLIENT_MAX_IDLE_CONNECTIONS);
+
+  /**
+   * Mechanism to override properties set in a static context.
+   */
+  private final Map<String, String> overrides;
+
+  /**
+   * Creates a new instance to handle Properties per instance.
+   */
+  public Properties() {
+    this.overrides = null;
+  }
+
+  /**
+   * Creates a new instance to handle Properties per instance.
+   * @param overrides to override static properties
+   */

```

Comment:
Could we make sure that we wrap the provided `overrides` parameter in `Collections.unmodifiableMap()` to avoid any surprises.